src/github/mod.rs

```diff
@@ -244,14 +244,46 @@ pub fn handle_issue(issue: IssueFromJson, repo: &str) -> DashResult<()> {
 }
 
 pub fn handle_user(user: &GitHubUser) -> DashResult<()> {
+    use diesel::pg::upsert::*;
+
     let conn = DB_POOL.get()?;
-    let exists = githubuser::table.find(user.id).get_result::<GitHubUser>(&*conn).is_ok();
+    diesel::insert(&user.on_conflict(githubuser::id, do_update().set(user)))
+        .into(githubuser::table).execute(&*conn)?;
+    Ok(())
+}
 
-    if exists {
-        diesel::update(githubuser::table.find(user.id)).set(user).execute(&*conn)?;
-    } else {
-        diesel::insert(user).into(githubuser::table).execute(&*conn)?;
-    }
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use dotenv;
 
-    Ok(())
+    #[test]
+    fn test_handle_user() {
```

Comment:
So I'm being pretty cautious right now and writing tests for everything, but this feels a little overkill -- all the functions essentially boil down to a single upsert statement now without any other logic, so assuming there aren't any bugs in diesel it should be pretty straightforward.

Thoughts?